tf-modules/tea-map-cache/main.tf

```
@@ -3,6 +3,20 @@ terraform {
     aws = ">= 2.31.0"
   }
 }
+
+data "aws_region" "current" {}
+
+resource "aws_vpc_endpoint" "config" {
+  count = var.deploy_to_ngap && (var.vpc_id == null || var.lambda_subnet_ids == null) ? 0 : 1
```

I'm sorry I missed this before (and with the caveat I don't have a non-NGAP account to really test it on): 
 

var.lambda_subnet_ids is a list of strings, with a default of `[]`, this should be checking it's length.  

Also - If deploy_to_ngap is true, *but* a user doesn't have vpc/subnets defined, this will evaluate to a count of 1 and attempt to deploy a vpc endpoint into NGAP.     I think we need to invert this so that in any situation `deploy_to_ngap` is true, it never tries to add the endpoint.  

How about 
```suggestion
  count = !var.deploy_to_ngap && (var.vpc_id == null || length(var.lambda_subnet_ids) != 0) ? 1 : 0
```

That gives you 

!T & (F) = 0
!T & (T) = 0
!F & (F) = 0
!F & (T) = 1

---

Which.....removes the entire reason for having this conditional.   🤔

---

Ah, that's true, `lambda_subnet_ids` is a list of strings. My question is, when deploying to NGAP, are the VPC endpoints already configured? In any case, I do think we need the inversion as you suggested because we would _only_ want the VPC endpoint to be created in the case that the user is deploying outside of NGAP and that they have vpc IDs defined, ie:
```
deploy_to_ngap = false
vpc_id is _not_ null
and length(lambda_subnet_ids) != 0

!false & (true || true) 
which evaluates to true & true = 1

or
deploy_to_ngap = false
vpc_id is _not_ null
and length(lambda_subnet_ids) == 0

!false & (true || false)
which evaluates to true & true = 1
```
Technically, we can create a vpc_endpoint without specifying subnet IDs but I think the deployment would fail anyway if there were no subnet IDs specified by the user so I am inclined to rewrite it as follows:
```
count = !var.deploy_to_ngap && (var.vpc_id == null && length(var.lambda_subnet_ids) != 0) ? 1 : 0
```

---

💯  that looks good.